packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -406,28 +405,23 @@ export const getChannelsNameMapInTeam: (state: GlobalState, teamId: string) => N
 );
 
 let prevChannelMap: NameMappedObjects<Channel> = {};
-let prevChannelDisplayNameMap: Dictionary<string> = {};
-
 export const getChannelsNameMapInCurrentTeam: (state: GlobalState) => NameMappedObjects<Channel> = createSelector(
     getAllChannels,
     getChannelSetInCurrentTeam,
     (channels: IDMappedObjects<Channel>, currentTeamChannelSet: string[]): NameMappedObjects<Channel> => {
         const channelMap: NameMappedObjects<Channel> = {};
-        const channelDisplayNameMap: Dictionary<string> = {};
         currentTeamChannelSet.forEach((id) => {
             const channel = channels[id];
             channelMap[channel.name] = channel;
-            channelDisplayNameMap[channel.name] = channel.display_name;
         });
 
         // getAllChannels will almost never return a memoized result.
         // Assumption here is that the added comparison computation is more beneficial
         // than the cost of returning a non memoized result.
-        if (deepEqual(channelDisplayNameMap, prevChannelDisplayNameMap)) {
+        if (shallowEquals(channelMap, prevChannelMap)) {
```

Comment:
The deep equality check isn't actually needed for these since channel objects are "immutable" when they come from the redux store